### PR TITLE
 camera plugin: add expand outer zoom limit option

### PR DESCRIPTION
**Pull request by MagicfTail**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java [hunk 1]
```diff
@@ -42,11 +42,22 @@ default boolean innerLimit()
 		return false;
 	}
 
+	@ConfigItem(
+		keyName = "outerLimit",
+		name = "Expand outer zoom limit",
+		description = "Configures whether or not the outer zoom limit is increased",
+		position = 2
+	)
+	default int outerLimit()
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java]: I'm assuming this is not meant to be an int? Either that, or the description is woefully unhelpful.

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 2]
```diff
@@ -81,6 +83,7 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 			return;
 		}
 
+		int outerZoomLimit = Math.min(zoomConfig.outerLimit(), OUTER_ZOOM_LIMIT);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: This would probably break horribly if too negative.

Comment (MagicfTail) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: The zoom resets at around -2500, so it never becomes a problem for me at least. Even then it's limited at -1k

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 3]
```diff
@@ -81,6 +83,7 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 			return;
 		}
 
+		int outerZoomLimit = Math.max(-zoomConfig.outerLimit(), OUTER_ZOOM_LIMIT);
```
Comment (Abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Maybe this should be `128 - zoomConfig.outerLimit()` so setting the config to 1 doesn't change the limit massively.

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 4]
```diff
@@ -95,6 +98,12 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 			return;
 		}
 
+		if ("outerZoomLimit".equals(event.getEventName()) && outerZoomLimit != 128)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Given that setting intStack values is cheap, is it important for us to have that `outerZoomLimit != 128` check? I'd like to see that check removed, or have `128` defined as a constant somewhere and not just be a magic number.

Comment (MagicfTail) [reply runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Removed the check

**Pull request merged**